sql/stringtype.go

```diff
@@ -272,7 +273,15 @@ func (t stringType) Convert(v interface{}) (interface{}, error) {
 			return "", nil
 		}
 
-		return s.ToString(nil)
+		str, err := s.ToString(nil)
```

Comment:
Should this case be handling all JSONValues, not just JSONDocuments?

Reply:
I just leave as it was, but makes sense. If we use `JSONValue` then we need to remove the following piece of code:
```go
if s.Val == nil {
	return "", nil
}
```
tests still pass without it.

Reply:
If we don't handle arbitrary JSONValues somehow in this function then JSONValues that aren't JSONDocuments won't be supported, right? Are users supposed to be able to use their own implementations of JSONValue?

Reply:
Changed! Also, it fixes another bug related to JSON null values. They were translated to "" instead of "null"